src/arkode/arkode.c

```diff
@@ -1883,6 +1922,38 @@ int arkInitialSetup(ARKodeMem ark_mem, sunrealtype tout)
     }
   }
 
+  /* Check that user has supplied an initial step size if fixedstep mode is on */
+  if ((ark_mem->fixedstep) && (ark_mem->hin == ZERO))
+  {
+    arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    "Fixed step mode enabled, but no step size set");
+    return (ARK_ILL_INPUT);
+  }
+
+  /* Test input tstop for legality (correct direction of integration) */
+  if (ark_mem->tstopset)
+  {
+    htmp = (ark_mem->h == ZERO) ? tout - ark_mem->tcur : ark_mem->h;
+    if ((ark_mem->tstop - ark_mem->tcur) * htmp <= ZERO)
+    {
+      arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                      MSG_ARK_BAD_TSTOP, ark_mem->tstop, ark_mem->tcur);
+      return (ARK_ILL_INPUT);
+    }
+  }
+
+  /* Check to see if y0 satisfies constraints */
+  if (ark_mem->constraintsSet)
+  {
+    conOK = N_VConstrMask(ark_mem->constraints, ark_mem->yn, ark_mem->tempv1);
+    if (!conOK)
+    {
+      arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                      MSG_ARK_Y0_FAIL_CONSTR);
+      return (ARK_ILL_INPUT);
+    }
+  }
+
```

Comment:
The `ewt` vector needs to be populated before `mriStep_Init` was called, since it is needed when setting initial step sizes for MRI calculations (when the user does not specify the initial step size).

Reply:
Ah, I see it's `step_init` that needed to be moved. These checks could probably stay where they were since there's no need to call `step_init` if they throw an error.